outputs.tf

```
@@ -22,6 +22,11 @@ output "private_dns_zone_name" {
   value       = try(module.private_dns[0].name, null)
 }
 
+output "public_dns_zone_name_servers" {
+  description = "Name servers of the public DNS zone"
+  value       = try(local.public_zone_name_servers, null)
```

null default is already handled at the local variable level

```suggestion
  value       = local.public_zone_name_servers
```